handler/core/src/main/java/io/knotx/fragments/task/factory/DefaultTaskFactory.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (C) 2019 Knot.x Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.knotx.fragments.task.factory;
+
+import io.knotx.fragments.api.Fragment;
+import io.knotx.fragments.engine.FragmentEventContext;
+import io.knotx.fragments.engine.Task;
+import io.knotx.fragments.engine.graph.Node;
+import io.knotx.fragments.task.TaskFactory;
+import io.knotx.fragments.task.exception.NodeFactoryNotFoundException;
+import io.knotx.fragments.task.exception.TaskNotFoundException;
+import io.knotx.fragments.task.factory.node.NodeFactory;
+import io.vertx.core.json.JsonObject;
+import io.vertx.reactivex.core.Vertx;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.ServiceLoader;
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+public class DefaultTaskFactory implements TaskFactory, NodeProvider {
+
+  public static final String NAME = "default";
+
+  private DefaultTaskFactoryConfig taskFactoryConfig;
+  private Map<String, NodeFactory> nodeFactories;
+
+  @Override
+  public String getName() {
+    return NAME;
+  }
+
+  @Override
+  public DefaultTaskFactory configure(JsonObject taskFactoryConfig, Vertx vertx) {
+    this.taskFactoryConfig = new DefaultTaskFactoryConfig(taskFactoryConfig);
+    nodeFactories = initFactories(vertx);
+    return this;
+  }
+
+  @Override
+  public boolean accept(FragmentEventContext eventContext) {
+    Fragment fragment = eventContext.getFragmentEvent().getFragment();
+    boolean fragmentContainsTask = fragment.getConfiguration()
+        .containsKey(taskFactoryConfig.getTaskNameKey());
+    return fragmentContainsTask && isTaskConfigured(eventContext);
+  }
+
+  private boolean isTaskConfigured(FragmentEventContext eventContext) {
+    // TODO
+    return true;
+  }
+
+  @Override
+  public Task newInstance(FragmentEventContext eventContext) {
+    Fragment fragment = eventContext.getFragmentEvent().getFragment();
+    String taskKey = taskFactoryConfig.getTaskNameKey();
+    String taskName = fragment.getConfiguration().getString(taskKey);
+
+    Map<String, TaskOptions> tasks = taskFactoryConfig.getTasks();
+    return Optional.ofNullable(tasks.get(taskName))
+        .map(taskOptions -> {
+          Node rootNode = initNode(taskOptions.getGraph());
```

Comment:
Please create method `initGraph` that will show explicitly that the whole graph is initialised here.